src/util/api.test.js

```diff
@@ -0,0 +1,279 @@
+          episode: ['https://rickandmortyapi.com/api/episode/8'],
+          url: 'https://rickandmortyapi.com/api/character/20',
+          created: '2017-11-04T22:34:53.659Z',
+        }),
+      };
+    }
+    case 'https://rickandmortyapi.com/api/location/5': {
+      return {
+        ok: true,
+        status: 200,
+        json: async () => ({
+          id: 5,
+          name: 'Anatomy Park',
+          type: 'Microverse',
+          dimension: 'Dimension C-137',
+          residents: [
+            'https://rickandmortyapi.com/api/character/12',
+            'https://rickandmortyapi.com/api/character/17',
+            'https://rickandmortyapi.com/api/character/96',
+            'https://rickandmortyapi.com/api/character/97',
+            'https://rickandmortyapi.com/api/character/98',
+            'https://rickandmortyapi.com/api/character/99',
+            'https://rickandmortyapi.com/api/character/100',
+            'https://rickandmortyapi.com/api/character/101',
+            'https://rickandmortyapi.com/api/character/108',
+            'https://rickandmortyapi.com/api/character/268',
+            'https://rickandmortyapi.com/api/character/300',
+          ],
+          url: 'https://rickandmortyapi.com/api/location/5',
+          created: '2017-11-10T13:08:46.060Z',
+        }),
+      };
+    }
+    case 'https://rickandmortyapi.com/api/episode/200': {
+      return {
+        ok: false,
+        status: 404,
+        json: async () => ({ error: 'Episode not found' }),
+      };
+    }
+    default: {
+      throw new Error(`Unmocked API endpoint: ${url}`);
+    }
+  }
+};
+
+describe('API fetching tests', () => {
+  beforeEach(() => {
+    window.fetch = jest.fn().mockImplementation(fetchMock);
+  });
+
+  afterEach(() => {
+    // @ts-ignore
+    global.fetch.mockClear();
+    delete global.fetch;
+  });
+
+  it('fetches all characters', async () => {
+    // Given
+    const pageCount = 1;
+
+    // When
+    const characterData = await API.fetchCharacters(pageCount);
+
+    // Then
+    expect(characterData).toHaveLength(1);
+    expect(characterData[0].name).toBe('Rick Sanchez');
+  });
+
+  it('fetches all locations', async () => {
+    // Given
+    const pageCount = 1;
+
+    // When
+    const locationData = await API.fetchLocations(pageCount);
+
+    // Then
+    expect(locationData).toHaveLength(1);
+    expect(locationData[0].name).toBe('Earth (C-137)');
+  });
+
+  it('fetches all episodes and ignores empty pages', async () => {
+    // Given
+    const pageCount = 1;
+
+    // When
+    const episodeData = await API.fetchEpisodes(pageCount);
+
+    // Then
+    expect(episodeData).toHaveLength(0);
+  });
+
+  it('fetches a character given an id', async () => {
+    // Given
+    const id = 20;
+
+    // When
+    const locationData = await API.fetchCharacterWithId(id);
+
+    // Then
+    expect(locationData.name).toBe('Ants in my Eyes Johnson');
```

Comment:
```suggestion
    // When
    const characterData = await API.fetchCharacterWithId(id);

    // Then
    expect(characterData.name).toBe('Ants in my Eyes Johnson');
```

Reply:
Faktycznie mi to uciekło, dzięki, poprawione.